packages/ed25519-signature-2018/src/Ed25519Signature2018.ts

```diff
@@ -157,15 +162,21 @@ export class Ed25519Signature2018 {
     }
 
     // ensure date is in string format
-    if (date !== undefined && typeof date !== "string") {
-      date = new Date(date).toISOString();
-      date = date.substr(0, date.length - 5) + "Z";
+    if (date && typeof date !== "string") {
+      if (date === undefined || date === null) {
+        date = new Date();
+      } else if (typeof date === "number" || typeof date === "string") {
+        date = new Date(date);
+      }
+      const str = date.toISOString();
+      date = str.substr(0, str.length - 5) + "Z";
```

Comment:
if this line ever happens there should be a warning... again, mutating method arguments should result in a warning or an error.